Importing necessary libraries

In [9]:
import numpy as np
import pandas as pd

We should load the preprocessed data so we will do the minimal steps that we did in notebook 02.

In [12]:
df = pd.read_csv("development.csv")

df["title"] = df["title"].fillna("")
df["article"] = df["article"].fillna("")

df["text"] = df["title"] + " " + df["article"]

X_text = df["text"]
y = df["label"]

again we should split train and validation sets like we did in notebook 02

In [13]:
from sklearn.model_selection import train_test_split

X_train_text, X_val_text, y_train, y_val = train_test_split(
    X_text,
    y,
    test_size=0.2,
    random_state=42,
    stratify=y
)

print(len(X_train_text), len(X_val_text))

63997 16000


so now we have the ratio 63997 to 16000 for train-validation sets

Also, like we did in 02, we should again turn our strings into numerical values for model to process

In [14]:
from sklearn.feature_extraction.text import TfidfVectorizer

vectorizer = TfidfVectorizer(
    lowercase=True,
    stop_words="english",
    max_features=20000
)

X_train = vectorizer.fit_transform(X_train_text)
X_val = vectorizer.transform(X_val_text)

print(X_train.shape, X_val.shape)

(63997, 20000) (16000, 20000)
